Build fails with recent change to CMakeLists.txt

**Description**
Attempting to build ROS2 version of Foxglove Bridge fails with various make errors.  We were  previously able to build on the same platform without problem.  

Appears to be related to latest commit which changed to clang as default compiler and added "-Wmost" compiler option.

- Version: 0.4.1
- Platform: Ubuntu 20.04 x86-64

**Steps To Reproduce**
Pulling latest code from repo and running `colcon build --symlink-install` produces the following output:
```
Starting >>> foxglove_bridge
--- stderr: foxglove_bridge                                                                                                                                   
c++: error: unrecognized command line option ‘-Wmost’
c++: error: unrecognized command line option ‘-Wmost’
make[2]: *** [CMakeFiles/foxglove_bridge.dir/build.make:63: CMakeFiles/foxglove_bridge.dir/ros2_foxglove_bridge/src/ros2_foxglove_bridge_node.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:163: CMakeFiles/foxglove_bridge.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
c++: error: unrecognized command line option ‘-Wmost’
c++: error: unrecognized command line option ‘-Wmost’
make[2]: *** [CMakeFiles/foxglove_bridge_base.dir/build.make:63: CMakeFiles/foxglove_bridge_base.dir/foxglove_bridge_base/src/foxglove_bridge.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/foxglove_bridge_base.dir/build.make:76: CMakeFiles/foxglove_bridge_base.dir/foxglove_bridge_base/src/parameter.cpp.o] Error 1
make[2]: *** [CMakeFiles/foxglove_bridge_base.dir/build.make:89: CMakeFiles/foxglove_bridge_base.dir/foxglove_bridge_base/src/serialization.cpp.o] Error 1
c++: error: unrecognized command line option ‘-Wmost’
make[2]: *** [CMakeFiles/foxglove_bridge_base.dir/build.make:102: CMakeFiles/foxglove_bridge_base.dir/foxglove_bridge_base/src/server_factory.cpp.o] Error 1
c++: error: unrecognized command line option ‘-Wmost’
make[2]: *** [CMakeFiles/foxglove_bridge_base.dir/build.make:115: CMakeFiles/foxglove_bridge_base.dir/foxglove_bridge_base/src/test/test_client.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:109: CMakeFiles/foxglove_bridge_base.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
---
Failed   <<< foxglove_bridge [0.56s, exited with code 2]

```

Removing "-Wmost" from line 20 of CMakeLists.txt and re-running colcon build succeeds without error.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,7 +17,7 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 macro(enable_strict_compiler_warnings target)
   target_compile_options(${target} PRIVATE
     $<$<CXX_COMPILER_ID:MSVC>:/WX /W4>
-    $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wall -Wextra -Wpedantic -Werror -Wmost -Wold-style-cast -Wfloat-equal>
+    $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wall -Wextra -Wpedantic -Werror -Wold-style-cast -Wfloat-equal>
   )
 endmacro()
 

